cython/gtsam/tests/test_logging_optimizer.py

```diff
@@ -43,6 +43,11 @@ def setUp(self):
         self.optimizer = gtsam.GaussNewtonOptimizer(
             graph, initial, self.params)
 
+        self.lmparams = gtsam.LevenbergMarquardtParams()
+        self.lmoptimizer = gtsam.LevenbergMarquardtOptimizer(
```

Comment:
I believe this can be done better.

I recommend a factory function that constructs the optimizer based on a flag or callable. 

Feel free to ignore and add a TODO if you're strapped for time.

Reply:
I believe we can deal with this later :)

Reply:
Be sure to add a TODO :)